martin-tile-utils/src/lib.rs

```diff
@@ -33,6 +36,37 @@ impl Display for TileCoord {
     }
 }
 
+impl TileCoord {
+    /// Checks provided coordinates for validity
+    /// before constructing [`TileCoord`] instance.
+    #[must_use]
+    pub fn checked(z: u8, x: u32, y: u32) -> Option<TileCoord> {
+        Self::are_valid(z, x, y).then_some(Self { z, x, y })
+    }
```

Comment:
if we add a checked constructor, we should likely also have an unchecked one

```suggestion
    /// Checks provided coordinates for validity
    /// before constructing [`TileCoord`] instance.
    ///
    /// Check [`Self::new_unchecked`] if you are sure that your inputs are possible.
    #[must_use]
    pub fn new_checked(z: u8, x: u32, y: u32) -> Option<TileCoord> {
        Self::are_valid(z, x, y).then_some(Self { z, x, y })
    }

    /// Constructs [`TileCoord`] instance from arguments without checking that the tiles can exist.
    ///
    /// Check [`Self::new_checked`] if you are unsure if your inputs are possible.
    #[must_use]
    pub fn new_unchecked(z: u8, x: u32, y: u32) -> Option<TileCoord> {
        Self { z, x, y }
    }
```